src/memory/process.rs

```diff
@@ -106,6 +107,104 @@ pub trait ProcessTraits where Self: Sized {
         buff: &mut [u8]
     ) -> Result<(), ProcessError>;
 
+    fn read_struct<T: Sized>(
+        &self,
+        addr: usize
+    ) -> Result<T, ProcessError> {
+        let mut buff = vec![0u8; size_of::<T>()];
+
+        let byte_buff = unsafe {
+            std::slice::from_raw_parts_mut(
+                buff.as_mut_ptr() as *mut u8,
+                buff.len()
+            )
+        };
+
+        self.read(addr, byte_buff.len(), byte_buff)?;
+
+        let s: T = unsafe { std::ptr::read(buff.as_ptr() as *const _) };
+
+        Ok(s)
```

Comment:
The heap allocation through the `Vec` is unfortunate but what's worse is that the `Vec`'s data might not be aligned properly w.r.t. `T`. Both should be fixed by this:
```rs
let mut uninit = std::mem::MaybeUninit::<T>::uninit();
let ptr: *mut u8 = uninit.as_mut_ptr().cast();
let slice = unsafe { std::slice::from_raw_parts_mut(ptr, std::mem::size_of::<T>()) };

self.read(addr, slice.len(), slice)?;

Ok(unsafe { uninit.assume_init() })
```
Might be a good idea to run [miri](https://github.com/rust-lang/miri) over it to make sure the unsafe code is valid.
Marking the method itself as unsafe might be warranted too since the unsafety invariant basically requires the generic `T` to be `repr(C)` which cannot be specified by trait bounds to the caller will have to promise.